routes/changeset_upload.js

```diff
@@ -13,7 +14,23 @@ module.exports = function (req, res, api, params, next) {
   var parts = params.id.split(':')
   var version = parts.length === 2 ? parts[1] : null
 
-  var r = req.pipe(osm2Obj({coerceIds: false}))
+  var mergeObservations = through.obj(function (doc, _, cb) {
+    api.getElement(doc.id, {version: doc.version}, function (err, old) {
+      if (err) return cb(err)
+      if (!old) {
+        return cb(new errors.NotFound('version #' + doc.version))
+      }
+
+      if (old.type === 'observation') {
+        doc.type = 'observation'
+        doc.tags = Object.assign({}, old.tags, doc.tags)
```

Comment:
Should we not be doing?

```js
const newDoc = Object.assign({}, old, doc, {type: 'observation'})
```

I don't think we need to merge tags? If the user deletes the tag in iD we would expect it to be deleted in MapFilter view no? I don't think iD changes tags without user intervention. But what is lost are the top-level properties: `attachments` , `metadata`, `schemaVersion`. Not sure if any others from https://github.com/digidem/mapeo-schema/blob/master/docs/observation.md ?

Reply:
I guess I thought about it the other way around. I was worried there would be schema issues here if the schema ever changes in the future, so enumerating all of the top-level properties here would increase complexity. Whereas this only depends on the attribute `tags` staying the same. It also begs the question, is iD supposed to be able to edit the other properties besides tags (e.g., lat, lon)?

Reply:
In chat yesterday, Gregor says it is OK if iD editor edits lat & lon of observations.